### PR TITLE
Mooncake store get zero copy meta optimization

### PR DESCRIPTION
**Pull request by huangtingwei9988**
## Motivation
In the get_buffer_meta method, `indices` is a CPU tensor. The performance of the for loop to obtain the item value of the CPU tensor is poor. `indices` are converted to list before the for loop starts, which can greatly improve performance

Test Scripts:
```
import torch
import time

def get_buffer_meta(data_ptr, indices, page_size, convert_to_list=False):
    if convert_to_list:
        indices = indices.tolist()
    ptr_list = []
    for index in range(0, len(indices), page_size):
        k_ptr = (
            data_ptr
            + indices[index]
            * 16
        )
        ptr_list.append(k_ptr)
    return ptr_list

host_indices = torch.arange(32768, dtype=torch.int64)

for i in range(100):
  start_time = time.time()
  a = get_buffer_meta(0, host_indices, 16)
  cost_time1 = time.time() - start_time
  start_time = time.time()
  a = get_buffer_meta(0, host_indices, 16, convert_to_list=True)
  cost_time2 = time.time() - start_time
  print(f"cost_time1:{cost_time1}, cost_time2:{cost_time2}")
```
result (test on L20):
```
cost_time1:0.009115934371948242, cost_time2:0.0011911392211914062
cost_time1:0.008185148239135742, cost_time2:0.00081634521484375
cost_time1:0.008499383926391602, cost_time2:0.0007359981536865234
cost_time1:0.00819087028503418, cost_time2:0.0007307529449462891
cost_time1:0.008219718933105469, cost_time2:0.0007336139678955078
cost_time1:0.008234024047851562, cost_time2:0.0007388591766357422
cost_time1:0.008252143859863281, cost_time2:0.0007562637329101562
cost_time1:0.008135795593261719, cost_time2:0.0007798671722412109
cost_time1:0.008404254913330078, cost_time2:0.0007798671722412109
cost_time1:0.008409261703491211, cost_time2:0.0007443428039550781
cost_time1:0.008191823959350586, cost_time2:0.0007863044738769531
cost_time1:0.008414268493652344, cost_time2:0.0007631778717041016
cost_time1:0.008271455764770508, cost_time2:0.0007350444793701172
cost_time1:0.033663034439086914, cost_time2:0.0008282661437988281
cost_time1:0.008255481719970703, cost_time2:0.0007886886596679688
cost_time1:0.008484125137329102, cost_time2:0.0007905960083007812
cost_time1:0.008503437042236328, cost_time2:0.0007834434509277344
cost_time1:0.008460283279418945, cost_time2:0.0007865428924560547
cost_time1:0.008426427841186523, cost_time2:0.0007369518280029297
cost_time1:0.008172035217285156, cost_time2:0.0007297992706298828
cost_time1:0.008249282836914062, cost_time2:0.0008311271667480469
cost_time1:0.008504629135131836, cost_time2:0.0007441043853759766
cost_time1:0.008193731307983398, cost_time2:0.0007979869842529297
cost_time1:0.008404970169067383, cost_time2:0.0007965564727783203
cost_time1:0.008534669876098633, cost_time2:0.0007898807525634766
cost_time1:0.00843191146850586, cost_time2:0.0008094310760498047
cost_time1:0.008510112762451172, cost_time2:0.0007381439208984375
cost_time1:0.008163690567016602, cost_time2:0.0007312297821044922
cost_time1:0.008129119873046875, cost_time2:0.0007982254028320312
cost_time1:0.00837850570678711, cost_time2:0.0007550716400146484
cost_time1:0.008196115493774414, cost_time2:0.0007255077362060547
cost_time1:0.008123397827148438, cost_time2:0.0007839202880859375
cost_time1:0.008397579193115234, cost_time2:0.0007562637329101562
cost_time1:0.008112907409667969, cost_time2:0.0007393360137939453
cost_time1:0.008367538452148438, cost_time2:0.0008442401885986328
cost_time1:0.008483171463012695, cost_time2:0.0007352828979492188
cost_time1:0.008258581161499023, cost_time2:0.0007407665252685547
cost_time1:0.008225202560424805, cost_time2:0.000732421875
cost_time1:0.008180618286132812, cost_time2:0.0007786750793457031
cost_time1:0.008456230163574219, cost_time2:0.0007319450378417969
cost_time1:0.00821542739868164, cost_time2:0.0007364749908447266
cost_time1:0.00829172134399414, cost_time2:0.0007870197296142578
cost_time1:0.008504629135131836, cost_time2:0.0007421970367431641
cost_time1:0.008206844329833984, cost_time2:0.0007860660552978516
cost_time1:0.008328914642333984, cost_time2:0.0007815361022949219
cost_time1:0.00843358039855957, cost_time2:0.0007410049438476562
cost_time1:0.008074045181274414, cost_time2:0.0007359981536865234
cost_time1:0.008175373077392578, cost_time2:0.0007877349853515625
cost_time1:0.00848531723022461, cost_time2:0.0007433891296386719
cost_time1:0.008185148239135742, cost_time2:0.0007381439208984375
cost_time1:0.008158683776855469, cost_time2:0.0007264614105224609
cost_time1:0.008171558380126953, cost_time2:0.0007281303405761719
cost_time1:0.008237361907958984, cost_time2:0.0007846355438232422
cost_time1:0.008458614349365234, cost_time2:0.0007369518280029297
cost_time1:0.008156299591064453, cost_time2:0.0007398128509521484
cost_time1:0.008145570755004883, cost_time2:0.0007870197296142578
cost_time1:0.00845026969909668, cost_time2:0.0007314682006835938
cost_time1:0.008229255676269531, cost_time2:0.0007817745208740234
cost_time1:0.008461475372314453, cost_time2:0.0007336139678955078
cost_time1:0.008262395858764648, cost_time2:0.0007300376892089844
cost_time1:0.00824880599975586, cost_time2:0.00074005126953125
cost_time1:0.008239269256591797, cost_time2:0.0008027553558349609
cost_time1:0.008559942245483398, cost_time2:0.0007302761077880859
cost_time1:0.00828695297241211, cost_time2:0.0007777214050292969
cost_time1:0.008524656295776367, cost_time2:0.0007476806640625
cost_time1:0.008248329162597656, cost_time2:0.000789642333984375
cost_time1:0.008495092391967773, cost_time2:0.0007359981536865234
cost_time1:0.008246660232543945, cost_time2:0.0007803440093994141
```
## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.ai/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.ai/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.ai/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.ai/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.ai/developer_guide/contribution_guide.html#benchmark-the-speed).

### CODE REVIEW

**Review by xiezhq-hermann — APPROVED** [review]

**Comment by ykwd** [comment]
I found that the impact of this optimization varies significantly across different GPUs. On the A10, the performance difference is not significant. However, on the H800, when len(indices) = 8192, this optimization reduces the execution time of a single `get_buffer_meta` call from 0.5 to 3 seconds down to just 1 to 2 ms.

**Comment by huangtingwei9988** [comment]
> I found that the impact of this optimization varies significantly across different GPUs. On the A10, the performance difference is not significant. However, on the H800, when len(indices) = 8192, this optimization reduces the execution time of a single `get_buffer_meta` call from 0.5 to 3 seconds down to just 1 to 2 ms.

@ykwd  I was surprised to see such an improvement on the H800！！
